code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -197,45 +199,68 @@
 		return attack_paw(M)
 
 /obj/structure/closet/attackby(obj/item/I, mob/user, params)
-	. = ..()
+	if(user in src)
+		return FALSE
+	if(tool_interact(I, user))
+		return TRUE // No afterattack
+	return ..()
+
 
+/obj/structure/closet/proc/tool_interact(obj/item/I, mob/user)
```

Comment:
This should be handled via the specific welder_act and such, and only for tools, not package wraps.